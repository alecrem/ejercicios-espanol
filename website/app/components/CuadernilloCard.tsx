interface CuadernilloCardProps {
  number: number;
  title: string;
  verbs: string[];
  description: string;
}

export default function CuadernilloCard({ number, title, verbs, description }: CuadernilloCardProps) {
  return (
    <div className="cuadernillo-card">
      <div className="cuadernillo-number">Cuadernillo {number}</div>
      <h3 className="cuadernillo-title">{title}</h3>
      <div className="cuadernillo-verbs">
        Verbos: {verbs.join(", ")}
      </div>
      <p className="cuadernillo-content">
        {description}
      </p>
      <div className="download-buttons">
        <a
          href={`/pdfs/japanese/cuadernillo-${number}-ja.pdf`}
          className="download-btn download-btn-primary"
          download
        >
          ðŸ“– Descargar en JaponÃ©s
        </a>
        <a
          href={`/pdfs/english/cuadernillo-${number}-en.pdf`}
          className="download-btn download-btn-secondary"
          download
        >
          ðŸ“– Descargar en InglÃ©s
        </a>
      </div>
    </div>
  );
}