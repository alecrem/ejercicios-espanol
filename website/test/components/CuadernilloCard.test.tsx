import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import CuadernilloCard from "~/components/CuadernilloCard";

describe("CuadernilloCard", () => {
  const mockProps = {
    number: 1,
    title: "primera conjugación (-AR)",
    verbs: ["dibujar", "buscar", "hablar", "bailar"],
    description: "Aprende los verbos de la primera conjugación"
  };

  it("displays cuadernillo number", () => {
    render(<CuadernilloCard {...mockProps} />);
    expect(screen.getByText("Cuadernillo 1")).toBeInTheDocument();
  });

  it("displays title", () => {
    render(<CuadernilloCard {...mockProps} />);
    expect(screen.getByText("primera conjugación (-AR)")).toBeInTheDocument();
  });

  it("displays list of verbs", () => {
    render(<CuadernilloCard {...mockProps} />);
    expect(screen.getByText("Verbos: dibujar, buscar, hablar, bailar")).toBeInTheDocument();
  });

  it("displays description", () => {
    render(<CuadernilloCard {...mockProps} />);
    expect(screen.getByText("Aprende los verbos de la primera conjugación")).toBeInTheDocument();
  });

  it("includes Japanese download link with correct href", () => {
    render(<CuadernilloCard {...mockProps} />);
    const japaneseLink = screen.getByRole("link", { name: "📖 Descargar en Japonés" });
    expect(japaneseLink).toBeInTheDocument();
    expect(japaneseLink).toHaveAttribute("href", "/pdfs/japanese/cuadernillo-1-ja.pdf");
    expect(japaneseLink).toHaveAttribute("download");
  });

  it("includes English download link with correct href", () => {
    render(<CuadernilloCard {...mockProps} />);
    const englishLink = screen.getByRole("link", { name: "📖 Descargar en Inglés" });
    expect(englishLink).toBeInTheDocument();
    expect(englishLink).toHaveAttribute("href", "/pdfs/english/cuadernillo-1-en.pdf");
    expect(englishLink).toHaveAttribute("download");
  });

  it("handles different cuadernillo numbers correctly", () => {
    render(<CuadernilloCard {...mockProps} number={3} />);
    
    expect(screen.getByText("Cuadernillo 3")).toBeInTheDocument();
    
    const japaneseLink = screen.getByRole("link", { name: "📖 Descargar en Japonés" });
    expect(japaneseLink).toHaveAttribute("href", "/pdfs/japanese/cuadernillo-3-ja.pdf");
    
    const englishLink = screen.getByRole("link", { name: "📖 Descargar en Inglés" });
    expect(englishLink).toHaveAttribute("href", "/pdfs/english/cuadernillo-3-en.pdf");
  });

  it("matches snapshot", () => {
    const { container } = render(<CuadernilloCard {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});