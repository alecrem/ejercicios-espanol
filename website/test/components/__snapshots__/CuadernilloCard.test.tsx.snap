// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CuadernilloCard > matches snapshot 1`] = `
<div>
  <div
    class="cuadernillo-card"
  >
    <div
      class="cuadernillo-number"
    >
      Cuadernillo 
      1
    </div>
    <h3
      class="cuadernillo-title"
    >
      primera conjugaciÃ³n (-AR)
    </h3>
    <div
      class="cuadernillo-verbs"
    >
      Verbos: 
      dibujar, buscar, hablar, bailar
    </div>
    <p
      class="cuadernillo-content"
    >
      Aprende los verbos de la primera conjugaciÃ³n
    </p>
    <div
      class="download-buttons"
    >
      <a
        class="download-btn download-btn-primary"
        download=""
        href="/pdfs/japanese/cuadernillo-1-ja.pdf"
      >
        ğŸ“– Descargar en JaponÃ©s
      </a>
      <a
        class="download-btn download-btn-secondary"
        download=""
        href="/pdfs/english/cuadernillo-1-en.pdf"
      >
        ğŸ“– Descargar en InglÃ©s
      </a>
    </div>
  </div>
</div>
`;
